### PR TITLE
refactor: result route

### PR DESCRIPTION
**Pull request by yash6102000**
result route refactor

### CODE REVIEW

File: app/components/services/results.service.ts [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+/**
+ *
+ * @param currentWorkspaceId
+ * @param resultsItemsPerPage
+ * @param resultsCurrentPage
+ * @param statusFilter
+ * @param sortBy
+ * @param sortOrder
+ * @returns this function will return all candidate tests
+ */
```
Comment (Dhruv-Copods) [inline app/components/services/results.service.ts:23]: We also need to write what the function is for and what its purpose is

File: app/components/services/results.service.ts [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+) => {
+  return getAllCandidateTests(
+    currentWorkspaceId as string,
+    resultsItemsPerPage,
+    resultsCurrentPage,
+    statusFilter,
+    sortBy as string,
+    sortOrder as string
+  )
+}
```
Comment (Dhruv-Copods) [inline app/components/services/results.service.ts:40]: We need to use Async Await

File: app/components/services/results.service.ts (hunk 1 above)
Comment (Dhruv-Copods) [reply app/components/services/results.service.ts:23]: Comment for whole file

File: app/routes/$workspaceId/results/groupByTests/index.tsx [hunk 3]
```diff
@@ -3,41 +3,41 @@ import { redirect } from "@remix-run/node"
...
+} from "~/components/services/results.service"
+import { sortByOrder } from "~/interface/Interface"
 type LoaderData = {
-  candidateTest: Awaited<ReturnType<typeof getAllCandidateTests>>
-  userId: Awaited<ReturnType<typeof getUserId>>
-  workspaces: Awaited<ReturnType<typeof getUserWorkspaces>>
+  candidateTest: any
+  userId: any
+  workspaces: any
   currentWorkspaceId: string
```
Comment (Dhruv-Copods) [inline app/routes/$workspaceId/results/groupByTests/index.tsx:18]: Why are we removing Awaited Return Type? Just exchange the function here

File: app/routes/$workspaceId/results/index.tsx [hunk 4]
```diff
@@ -1,31 +1,32 @@
...
-import { getUserWorkspaces } from "~/models/workspace.server"
-import { getUserId } from "~/session.server"
 
 type LoaderData = {
-  candidateTest: Awaited<ReturnType<typeof getAllCandidateTests>>
-  userId: Awaited<ReturnType<typeof getUserId>>
-  workspaces: Awaited<ReturnType<typeof getUserWorkspaces>>
+  candidateTest: any
+  userId: string
+  workspaces: any
```
Comment (Dhruv-Copods) [inline app/routes/$workspaceId/results/index.tsx:15]: Same here we need the Awaited Return Type

File: app/routes/$workspaceId/results/groupByTests/$testId.tsx [hunk 5]
```diff
@@ -2,32 +2,31 @@ import type { ActionFunction } from "@remix-run/node"
...
-  getAllCandidatesOfTestCount,
-} from "~/models/result.server"
-import { getUserWorkspaces } from "~/models/workspace.server"
-import { getUserId } from "~/session.server"
+  getALLCandidatesOfTest,
+  getALLCandidatesOfTestCount,
+  getTestResendLink,
+  getUsersId,
+  getWorkspaces,
+} from "~/components/services/results.service"
```
Comment (Dhruv-Copods) [inline app/routes/$workspaceId/results/groupByTests/$testId.tsx:12]: Import directory is wrong we need to create service folder in app folder, ./app/services

File: app/routes/$workspaceId/results/groupByTests/index.tsx [hunk 6]
```diff
@@ -3,41 +3,41 @@ import { redirect } from "@remix-run/node"
...
+  getWorkspaces,
+} from "~/components/services/results.service"
+import { sortByOrder } from "~/interface/Interface"
 type LoaderData = {
-  candidateTest: Awaited<ReturnType<typeof getAllCandidateTests>>
-  userId: Awaited<ReturnType<typeof getUserId>>
-  workspaces: Awaited<ReturnType<typeof getUserWorkspaces>>
+  candidateTest: any
+  userId: any
+  workspaces: any
```
Comment (Dhruv-Copods) [inline app/routes/$workspaceId/results/groupByTests/index.tsx:17]: Why are we removing Awaited Return Type? Just exchange the function here

File: app/routes/$workspaceId/results/index.tsx [hunk 7]
```diff
@@ -1,31 +1,32 @@
 import type { LoaderFunction } from "@remix-run/node"
 import { redirect } from "@remix-run/node"
 import { json } from "@remix-run/node"
-
 import GroupByTests from "~/components/results/GroupByTests"
+import {
+  getALLCandidateTests,
+  getUsersId,
+  getWorkspaces,
+} from "~/components/services/results.service"
```
Comment (Dhruv-Copods) [inline app/routes/$workspaceId/results/index.tsx:9]: Update file structure

File: app/routes/$workspaceId/results/groupByTests/$testId.tsx [hunk 8]
```diff
@@ -55,11 +55,11 @@ export const action: ActionFunction = async ({ request }) => {
     const testId = formData.get("testId") as string
     const candidateId = formData.get("candidateId") as string
     const id = formData.get("id") as string
-    const candidateInviteStatus = await resendTestLink({
+    const candidateInviteStatus = await getTestResendLink(
```
Comment (anuragP04) [inline app/routes/$workspaceId/results/groupByTests/$testId.tsx:58]: don't update function argument type

**Review by anuragP04 — APPROVED** [review]

**Pull request merged**